prov/verbs/src/verbs_info.c

```diff
@@ -169,199 +164,30 @@ int fi_ibv_check_ep_attr(const struct fi_ep_attr *attr,
 		return -FI_ENODATA;
 	}
 
-	if (attr->protocol_version > 1) {
-		VERBS_INFO(FI_LOG_CORE,
-			   "Unsupported protocol version\n");
-		return -FI_ENODATA;
-	}
-
-	if (attr->max_msg_size > info->ep_attr->max_msg_size) {
-		VERBS_INFO(FI_LOG_CORE,
-			   "Max message size too large\n");
-		FI_INFO_CHECK_VAL(&fi_ibv_prov, info->ep_attr, attr,
-				  max_msg_size);
-		return -FI_ENODATA;
-	}
-
-	if (attr->max_order_raw_size > info->ep_attr->max_order_raw_size) {
-		VERBS_INFO( FI_LOG_CORE,
-			   "max_order_raw_size exceeds supported size\n");
-		FI_INFO_CHECK_VAL(&fi_ibv_prov, info->ep_attr, attr,
-				  max_order_raw_size);
-		return -FI_ENODATA;
-	}
-
-	if (attr->max_order_war_size) {
-		VERBS_INFO(FI_LOG_CORE,
-			   "max_order_war_size exceeds supported size\n");
-		FI_INFO_CHECK_VAL(&fi_ibv_prov, info->ep_attr, attr,
-				  max_order_war_size);
-		return -FI_ENODATA;
-	}
-
-	if (attr->max_order_waw_size > info->ep_attr->max_order_waw_size) {
-		VERBS_INFO(FI_LOG_CORE,
-			   "max_order_waw_size exceeds supported size\n");
-		FI_INFO_CHECK_VAL(&fi_ibv_prov, info->ep_attr, attr,
-				  max_order_waw_size);
-		return -FI_ENODATA;
-	}
-
-	if (attr->tx_ctx_cnt > info->domain_attr->max_ep_tx_ctx) {
-		VERBS_INFO(FI_LOG_CORE,
-			   "tx_ctx_cnt exceeds supported size\n");
-		VERBS_INFO(FI_LOG_CORE, "Supported: %zd\nRequested: %zd\n",
-			   info->domain_attr->max_ep_tx_ctx, attr->tx_ctx_cnt);
-		return -FI_ENODATA;
-	}
-
-	if ((attr->rx_ctx_cnt > info->domain_attr->max_ep_rx_ctx) &&
-	    (attr->rx_ctx_cnt != FI_SHARED_CONTEXT)) {
-		VERBS_INFO(FI_LOG_CORE,
-			   "rx_ctx_cnt exceeds supported size\n");
-		VERBS_INFO(FI_LOG_CORE, "Supported: %zd\nRequested: %zd\n",
-			   info->domain_attr->max_ep_rx_ctx, attr->rx_ctx_cnt);
-		return -FI_ENODATA;
-	}
-
-	if (attr->auth_key_size &&
-	    (attr->auth_key_size != info->ep_attr->auth_key_size)) {
-		VERBS_INFO(FI_LOG_CORE, "Unsupported authentification size.");
-		FI_INFO_CHECK_VAL(&fi_ibv_prov, info->ep_attr, attr,
-				  auth_key_size);
-		return -FI_ENODATA;
-	}
-
-	return 0;
+	return ofi_check_ep_attr(&fi_ibv_util_prov,
+				 info->fabric_attr->api_version,
+				 info, &user_attr);
```

Comment:
Thanks for removing the redundant check functions!

Reply:
You're welcome. I guess, it helps to avoid redundant updates whenever the attributes of the `fi_info` is changed